File: backend/app/schemas/messaging.py
import uuid
from datetime import datetime
from pydantic import BaseModel
from app.schemas.user import UserSummary


class ThreadCreate(BaseModel):
    subject: str
    recipient_ids: list[uuid.UUID]
    course_id: uuid.UUID | None = None
    initial_message: str


class MessageCreate(BaseModel):
    body: str


class MessageRead(BaseModel):
    id: uuid.UUID
    sender_id: uuid.UUID
    body: str
    sent_at: datetime

    model_config = {"from_attributes": True}


class ThreadRead(BaseModel):
    id: uuid.UUID
    subject: str
    course_id: uuid.UUID | None
    created_by: uuid.UUID
    unread_count: int = 0
    last_message: MessageRead | None = None
    participants: list[UserSummary] = []
    created_at: datetime

    model_config = {"from_attributes": True}


class ThreadDetailRead(ThreadRead):
    messages: list[MessageRead]

File: frontend/src/types/messaging.ts
export interface Message {
  id: string
  sender_id: string
  body: string
  sent_at: string
}

export interface Thread {
  id: string
  subject: string
  course_id: string | null
  created_by: string
  unread_count: number
  last_message: Message | null
  participants: { id: string; full_name: string; avatar_url: string | null }[]
  created_at: string
  messages?: Message[]
}
